internal/errcheck/errcheck.go

```diff
@@ -175,28 +172,18 @@ func (c *Checker) logf(msg string, args ...interface{}) {
 	}
 }
 
-func (c *Checker) load(paths ...string) (*loader.Program, error) {
-	ctx := build.Default
-	for _, tag := range c.Tags {
-		ctx.BuildTags = append(ctx.BuildTags, tag)
-	}
-	loadcfg := loader.Config{
-		Build: &ctx,
-	}
-
-	if c.WithoutGeneratedCode {
-		loadcfg.ParserMode = parser.ParseComments
-	}
+// loadPackages is used for testing.
+var loadPackages = func(cfg *packages.Config, paths ...string) ([]*packages.Package, error) {
+	return packages.Load(cfg, paths...)
+}
 
-	rest, err := loadcfg.FromArgs(paths, !c.WithoutTests)
-	if err != nil {
-		return nil, fmt.Errorf("could not parse arguments: %s", err)
-	}
-	if len(rest) > 0 {
-		return nil, fmt.Errorf("unhandled extra arguments: %v", rest)
+func (c *Checker) load(paths ...string) ([]*packages.Package, error) {
+	cfg := &packages.Config{
+		Mode:       packages.LoadAllSyntax,
+		Tests:      !c.WithoutTests,
+		BuildFlags: []string{fmt.Sprintf("-tags=%s", strings.Join(c.Tags, " "))},
```

Comment:
Does this actually work with multiple tags? I feel like it would need to be quoted. Is this even needed anymore?

Reply:
We definitely should continue to support build tags, I don't know if this works or not as there are no tests for the functionality.

Reply:
Quoting is for shells, not argv. This should work, but I didn't test the PR.

Reply:
This produces a single string though, not a set of strings (which would make sense to do)

Reply:
BuildFlags is an array of arguments to pass to the `go` tool, in addition to the flags go/packages generates. Same semantics as os.Args, exec.Command or execve.

`go list -tags="foo bar"` becomes

```
([]string) (len=3 cap=3) {
 (string) (len=5) "go",
 (string) (len=4) "list",
 (string) (len=13) "-tags=foo bar"
}
```

Reply:
Correct. However, above, the length of BuildFlags is always 1. It is a single string, courtesy of fmt.Sprintf

Reply:
Never mind, I get it now. Carry on.

Reply:
I've put up a PR that adds tests for this at https://github.com/kisielk/errcheck/pull/160.

I applied the test to this PR (with the necessary modifications: see https://github.com/coopernurse/errcheck/pull/1) and verified that this seems to work as expected.

Reply:
Fabulous. I merged your tests in and they're now included in this PR (which was rebased off master today).